warp10/src/main/java/io/warp10/script/binary/ADD.java

```diff
@@ -137,13 +137,16 @@ public Object op(GeoTimeSerie gtsa, GeoTimeSerie gtsb, int idxa, int idxb) {
           };
           break;
         default:
-          // both inputs are empty, return an empty gts.
-          stack.push(new GeoTimeSerie());
-          return stack;
+          // Leave op to null.
```

Comment:
There should be a test in applyBinaryOp that the operator is non null, as is the op.op() call will fail with an NPE.

Reply:
I agree this would be nicer, but this won't lead to a NPE because the default means both GTSs are empty and `applyBinaryOp` won't apply `op` at all.